collection-scripts/gather_etcd

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="/must-gather"
+ETCD_LOG_PATH="${BASE_COLLECTION_PATH}/etcd_info"
+
+ETCDCTL_CONTAINER='etcdctl'
+
+RUNNING_ETCD_POD=$(oc get pod -n openshift-etcd --no-headers|grep Running| grep -o -m 1 '\S*etcd\S*')
+
+# We cannot rely on ETCDCTL_ENDPOINTS on container because it may contain bootstrap VM
+ETCDCTL_ENDPOINTS=$(oc exec ${RUNNING_ETCD_POD}  -n openshift-etcd -c ${ETCDCTL_CONTAINER} -- /bin/sh -c "etcdctl member list" | awk -F', ' '{printf "%s%s",sep,$5; sep=","}')
```

Comment:
There are many missing quotes here, and through the rest of this script, that would fail ShellCheck:

```console
$ shellcheck collection-scripts/gather_etcd 

In collection-scripts/gather_etcd line 11:
ETCDCTL_ENDPOINTS=$(oc exec ${RUNNING_ETCD_POD}  -n openshift-etcd -c ${ETCDCTL_CONTAINER} -- /bin/sh -c "etcdctl member list" | awk -F', ' '{printf "%s%s",sep,$5; sep=","}')
                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In collection-scripts/gather_etcd line 14:
    oc -n openshift-etcd exec -c ${ETCDCTL_CONTAINER} ${RUNNING_ETCD_POD} -- env ETCDCTL_ENDPOINTS=${ETCDCTL_ENDPOINTS} etcdctl $@
                                                      ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                                   ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                                                                ^-- SC2068: Double quote array expansions, otherwise they're like $* and break on spaces.


In collection-scripts/gather_etcd line 60:
wait ${PIDS[@]}
     ^-- SC2068: Double quote array expansions, otherwise they're like $* and break on spaces.

```

I think we should grow a ShellCheck job as a preflight for this repo.

Reply:
Let me fix that on `etcd_gather` script.

And I totally agree there should be an automated `ShellCheck` test, because none of the collection scripts completely passes it. It seems that there are quotes missing in many places. But I think I should just fix the etcd script at this PR, and fixing the other scripts should be another PR.